bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryResult.java

```diff
@@ -80,12 +80,20 @@
      * discovered. Its actual value can be retrieved from the {@link DiscoveryResult#getProperties()} map. Such unique
      * identifiers are typically the <code>ipAddress</code>, the <code>macAddress</code> or the
      * <code>serialNumber</code> of the discovered thing.
-     * 
+     *
      * @return the representation property of this result object (could be null)
      */
     public String getRepresentationProperty();
 
     /**
+     * Returns the actual value of the representation property of the result object.
+     *
+     * @see #getRepresentationProperty()
+     * @return the value of the representation property of this result object (could be null)
+     */
+    public Object getRepresentationPropertyValue();
```

Comment:
On a second look I must disagree. Removing the existing method getRepresentationProperty would break not only the JSON of the inbox events (at least change it), but also make the method DiscoveryResultImpl#synchronize impossible as I would have no way to access the representationProperty name to copy it into the new class. 

What about leaving everything as it is and only renaming my method getRepresentationPropertyValue() into getRepresentationValue()?